openverse_catalog/dags/providers/provider_api_scripts/nappy.py

```diff
@@ -0,0 +1,123 @@
+"""
+Content Provider:       Nappy
+
+ETL Process:            Use the API to identify all CC0-licensed images.
+
+Output:                 TSV file containing the image meta-data.
+
+Notes:                  This api was written specially for Openverse.
+                        There are no known limits or restrictions.
+
+"""
+import logging
+
+from common import constants
+from common.licenses import get_license_info
+from common.loader import provider_details as prov
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
+
+
+logger = logging.getLogger(__name__)
+
+
+class NappyDataIngester(ProviderDataIngester):
+    providers = {constants.IMAGE: prov.NAPPY_DEFAULT_PROVIDER}
+    endpoint = "https://api.nappy.co/v1/openverse/images"
+    headers = {"User-Agent": prov.UA_STRING, "Accept": "application/json"}
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        # On the first request, `prev_query_params` will be `None`. We can detect this
+        # and return our default params.
+        if not prev_query_params:
+            return {
+                "page": 1,
+            }
+        else:
+            return {
+                "page": prev_query_params["page"] + 1,
+            }
```

Comment:
After running this locally, it seems like we're getting 10 images back per response. I played around with the API a bit and found that we can use the `per_page` parameter to request a different number of images! We should definitely use `batch_limit` here to speed up ingestion :smile:

```suggestion
        if not prev_query_params:
            return {
                "page": 1,
                "per_page": self.batch_limit,
            }
        else:
            return {
                **prev_query_params,
                "page": prev_query_params["page"] + 1,
            }
```